source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -189,7 +191,16 @@ namespace RendererQml
                     if (Utils::IsInstanceOfSmart<AdaptiveCards::ShowCardAction>(actions[i]))
                     {
                         const auto showCardAction = std::dynamic_pointer_cast<AdaptiveCards::ShowCardAction>(actions[i]);
-                        //TODO: Add show card logic
+                        
+						auto subContext = std::make_shared<AdaptiveRenderContext>(context->GetConfig(), context->GetElementRenderers());
+						auto uiCard = subContext->Render(showCardAction->GetCard(), &AdaptiveCardRender);
+
+						if (uiCard != nullptr)
+						{
+							uiCard->Property("readonly property int margins", "0");
+							uiCard->Property("width", Formatter() << "600 - " << std::to_string(context->GetConfig()->GetSpacing().paddingSpacing*2));
```

Comment:
just a quick confirmation, you wanted the width to be calculate in this manner right ?

_this is width of showcard's inner rectangle where we subtract the margins_